control-plane/agents/src/bin/core/volume/snapshot_operations.rs

```diff
@@ -283,4 +358,35 @@ impl OperationGuardArc<VolumeSnapshot> {
 
         Ok(VolumeSnapshotCreateResult::new(replica_snap.1, timestamp))
     }
+
+    pub(crate) async fn destroy_replica_snapshot(
```

Comment:
doesn't need to be pub at all?

Reply:
It needs to be otherwise would not be able to use in the reconciler.

Reply:
why would the reconciler call this function directly?

Reply:
Because its a part of the `OperationGuardArc<VolumeSnapshot>`?

Reply:
Wouldn't the reconciler just call snapshot.destroy() ?

Reply:
Ohh right, sorry I mistook the  snapshot.destroy()  for this method. :-p I will change this one.